core/app/[locale]/(default)/account/(tabs)/order/[slug]/page-data.tsx

```diff
@@ -0,0 +1,199 @@
+    shippingMethodName
+    shippingProviderName
+    tracking {
+      __typename
+      ... on OrderShipmentNumberAndUrlTracking {
+        number
+        url
+      }
+      ... on OrderShipmentUrlOnlyTracking {
+        url
+      }
+      ... on OrderShipmentNumberOnlyTracking {
+        number
+      }
+    }
+  }
+`);
+
+const CustomerOrderDetails = graphql(
+  `
+    query CustomerOrderDetails($filter: OrderFilterInput) {
+      site {
+        order(filter: $filter) {
+          entityId
+          orderedAt {
+            utc
+          }
+          status {
+            label
+            value
+          }
+          totalIncTax {
+            value
+            currencyCode
+          }
+          subTotal {
+            value
+            currencyCode
+          }
+          discounts {
+            nonCouponDiscountTotal {
+              value
+              currencyCode
+            }
+            couponDiscounts {
+              couponCode
+              discountedAmount {
+                value
+                currencyCode
+              }
+            }
+          }
+          shippingCostTotal {
+            value
+            currencyCode
+          }
+          taxTotal {
+            value
+            currencyCode
+          }
+          billingAddress {
+            firstName
+            lastName
+            address1
+            city
+            stateOrProvince
+            postalCode
+            country
+          }
+          consignments {
+            shipping {
+              edges {
+                node {
+                  entityId
+                  shippingAddress {
+                    firstName
+                    lastName
+                    address1
+                    city
+                    stateOrProvince
+                    postalCode
+                    country
+                  }
+                  shipments {
+                    edges {
+                      node {
+                        entityId
+                        shippedAt {
+                          utc
+                        }
+                        ...OrderShipmentFragment
+                      }
+                    }
+                  }
+                  lineItems {
+                    edges {
+                      node {
+                        ...OrderItemFragment
+                      }
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  `,
+  [OrderItemFragment, OrderShipmentFragment],
+);
+
+export const getProductAttributes = cache(async (variables: ProductAttributesVariables) => {
+  const response = await client.fetch({
+    document: ProductAttributes,
+    variables,
+    fetchOptions: { next: { revalidate } },
+  });
+
+  return response.data.site.product;
+});
+
+type ShippingConsignments = NonNullable<
+  NonNullable<ResultOf<typeof CustomerOrderDetails>['site']['order']>['consignments']
+>['shipping'];
```

Comment:
Can we use `ExistingResultType`?

Reply:
As I answered in other place this code will be refactored (removed) in upcoming PR with Suspense for Orders. For that reason I haven't changed it here. 
I decided to split changes in 2 PRs. This one is adding slug for `OrderDetails` page. 2nd PR will add Suspense  & loading state for ProductSnippet and removes extra code after that.